mod/missions/pages/missions/display-search-set.php

```diff
@@ -95,6 +96,66 @@
 
 $content .= elgg_view('page/elements/mission-tabs') . $max_reached;
 
+if ($search_form ) {
+    
+	$content .= '<div style="clear:both">'.elgg_view_form('missions/search-simple').'</div>';
+}
+
+ $search_fields = array(
+	        //'' => 'Choose',
+	        '',
+	        elgg_echo('missions:title'),
+	        elgg_echo('missions:type'),
+	        elgg_echo('missions:department'),
+	        elgg_echo('missions:key_skills'),
+	        elgg_echo('missions:security_clearance'),
+	        elgg_echo('missions:location'),
+	        elgg_echo('missions:language'),
+	        elgg_echo('missions:time'),
+	        elgg_echo('missions:period'),
+	        elgg_echo('missions:start_time'),
+	        elgg_echo('missions:duration'),
+	        elgg_echo('missions:work_remotely'),
+	        elgg_echo('missions:program_area')
+    );
+
+if ($advanced_form){
+    $content .='<strong>'.elgg_echo('missions:search_value').':</strong> ';
+	for ($s = 0; $s < $number_of_rows; $s ++) {
+        if ($advanced_form['selection_'.$s]){
+
+        	$value =str_replace('missions:', '', $advanced_form['selection_'.$s.'_element']);
+
+           $content .= '<span class="mrgn-rght-md"><strong>'.$advanced_form['selection_'.$s].':</strong> '.$value.' </span> ';
```

Comment:
When the value is echoed here from a select option it will need to be put through an elgg_echo to display properly. Right now Micro-Mission displays as micro_mission because this is the value of the input.